chess/ChessGameViewModel.swift

```diff
@@ -9,41 +9,45 @@ import SwiftUI
 
 final class ChessGameViewModel: ObservableObject {
 
-    @Published var selectedPieceAddress: (row: Int, col: Int)?
+    @Published var selectedPieceAddress: Coordinate?
     @Published private var model = ChessGameModel()
 
     func gameAppeared() {
         self.model.createNewGameBoard()
     }
 
-    func selectPieceAt(row: Int, col: Int) {
-        guard let selectedPieceAddress = self.selectedPieceAddress else {
-            self.selectedPieceAddress = (row, col)
-            return
-        }
-
-        if selectedPieceAddress == (row, col) {
+    func handleTap(at cellAddress: Coordinate) {
+        switch (self.model.board[cellAddress.row][cellAddress.col] == nil, self.selectedPieceAddress == nil) {
+        case (true, false):
+            print("empty cell tapped and we have selected piece")
+            self.model.move(from: self.selectedPieceAddress!, to: cellAddress)
             self.selectedPieceAddress = nil
-        }
-        else {
-            self.selectedPieceAddress = (row, col)
+        case (false, true):
+            print("non-empty cell tapped and we don't have selected piece")
+            self.selectedPieceAddress = cellAddress
+        case (false, false):
+            print("non-empty cell tapped and we have selected piece")
+            self.selectedPieceAddress = cellAddress
+        case (true, true):
+            print("empty cell tapped and we don't have selected piece")
+            return
         }
     }
 
-    func getPieceForCell(row: Int, col: Int) -> ChessPiece? {
-        self.model.board[row][col]
+    func getPiece(for cell: Coordinate) -> ChessPiece? {
```

Comment:
let's unify naming for parameters that have `Coordinate` type. Here it is a `cell` but down below it is `address`. I assume word `cell` is recerved by the logic so coordinate(s) or address would make more sense